crypto/asn1/a_d2i_fp.c

```diff
@@ -92,7 +91,7 @@ void *ASN1_item_d2i_fp(const ASN1_ITEM *it, FILE *in, void *x)
 
 #define HEADER_SIZE   8
 #define ASN1_CHUNK_INITIAL_SIZE (16 * 1024)
-static int asn1_d2i_read_bio(BIO *in, BUF_MEM **pb)
+int asn1_d2i_read_bio(BIO *in, BUF_MEM **pb)
```

Comment:
And I think if you want a function that does it then it should be promoted into the API - rather than used in this manner - documented - and available - to me that is part of noting a gap in the API when you want to do this sort of things. And to me the name of that function is entirely incorrect. And changing ASN1_d2i_bio so it allowed for a NULL d2i parameter would have got you the access. That would be a more sensible change to me at least. Although it would need further tweaks. But renaming this to ASN1_d2i_read_bio makes it clear it is now exposed.